app/src/main/java/com/github/se/cyrcle/io/MapTileStorage.kt

```diff
@@ -0,0 +1,46 @@
+package com.github.se.cyrcle.io
+
+import java.io.File
+import java.io.IOException
+
+private const val CACHE_DIR = "./cache/map"
+private const val EXTENSION_NAME = "mapbin"
+
+/**
+ * Store a tile at the specified zoom level and coordinates.
+ *
+ * @param zoomLevel The zoom level of the tile.
+ * @param x The x-coordinate of the tile.
+ * @param y The y-coordinate of the tile.
+ * @param tileData The byte array of the tile data.
+ * @throws IOException If an I/O error occurs.
+ */
+@Throws(IOException::class)
+fun storeTile(zoomLevel: Int, x: Int, y: Int, tileData: ByteArray) {
+  val tileDir = File("$CACHE_DIR/$zoomLevel/$x")
+  if (!tileDir.exists()) {
+    tileDir.mkdirs()
+  }
+  val tileFile = File(tileDir, "$y.$EXTENSION_NAME")
+  if (tileFile.exists()) return // Tile should change, there is no point in updating them
```

Comment:
Indeed, this seems to be more simpler and less error-prone. I'll even modify it to allow update with some overwrite argument, so that usage is more flexible and logic more readable.